core-common/src/main/java/org/uniprot/core/util/EnumDisplay.java

```diff
@@ -32,29 +41,45 @@
     }
 
     /**
-     * ignore case and trim on passed string argument to match
+     * Returns the Enum constant by its name. String value of name is trimmed and case is ignored
+     * while finding the corresponding enum constant.
      *
-     * @throws IllegalArgumentException for null or not matching argument
+     * @param name string value of the enum to be returned.
+     * @param enumClass the Class object of the enum type
+     * @return Enum constant for the given name and enum type
+     * @throws IllegalArgumentException if either name or enumClass is null. or name doesn't have
+     *     corresponding enum
      */
-    static <T extends Enum<T> & EnumDisplay> T typeOf(@Nonnull String name, Class<T> enumClass) {
+    static <T extends Enum<T> & EnumDisplay> T typeOf(
+            @Nonnull String name, @Nonnull Class<T> enumClass) {
         nullThrowIllegalArgument(name);
+        nullThrowIllegalArgument(enumClass);
 
-        for (T constant : enumClass.getEnumConstants()) {
-            if (constant.getCompareOn().equalsIgnoreCase(name.trim())) return constant;
-        }
-        throw new IllegalArgumentException(
-                "The " + enumClass.getSimpleName() + " with " + name + " doesn't exist");
+        RuntimeException exception =
+                new IllegalArgumentException(
+                        "The " + enumClass.getSimpleName() + " with " + name + " doesn't exist");
+
+        return Arrays.stream(enumClass.getEnumConstants())
+                .filter(constant -> constant.getCompareOn().equalsIgnoreCase(name.trim()))
+                .findAny()
+                .orElseThrow(() -> exception);
     }
 
     /**
-     * ignore case and trim on passed string argument to match
+     * Returns the Enum constant by its name or given defaultValue in case of no match.
      *
-     * @return matching value or defaultValue in case of no match
+     * @param name string value of the enum to be returned.
+     * @param enumClass the Class object of the enum type
+     * @param defaultValue the default enum constant in case of no enum constant by given name and
+     *     type
+     * @return Enum constant for the given name and enum type Or defaultValue in case of i. no
+     *     match, ii. name is null or iii. enumClass is null
      */
-    static <T extends Enum<T> & EnumDisplay> T typeOf(@Nonnull String name, T defaultValue) {
+    static <T extends Enum<T> & EnumDisplay> T typeOf(
+            @Nonnull String name, @Nonnull Class<T> enumClass, @Nullable T defaultValue) {
```

Comment:
i wonder, why would you add third parameter into this method, two was enough and doing job 
```suggestion
    static <T extends Enum<T> & EnumDisplay> T typeOf(
            @Nonnull String name, @Nonnull Class<T> enumClass, @Nullable T defaultValue) {
```
do we have any usecase where we want null as default value?
```suggestion
    static <T extends Enum<T> & EnumDisplay> T typeOf(
            @Nonnull String name, @Nonnull Class<T> enumClass, @Nullable T defaultValue) {
```